assets/src/stories-editor/components/block-preview-label.js

```diff
@@ -76,19 +79,19 @@ const BlockPreviewLabel = ( { block, label, displayIcon = true, alignIcon = 'lef
 				const dateFormat = dateSettings.formats.date;
 				const date = postDate || new Date();
 
-				label = dateI18n( dateFormat, date );
+				blockPreviewLabel.current = dateI18n( dateFormat, date );
 				break;
 
 			case 'amp/amp-story-post-title':
-				label = getEditedPostAttribute( 'title' ) || blockType.title;
+				blockPreviewLabel.current = getEditedPostAttribute( 'title' ) || blockType.title;
 				break;
 
 			default:
 				break;
 		}
 
 		return {
-			content: label,
+			content: blockPreviewLabel.current,
 			icon: blockType.icon,
 		};
 	}, [ block ] );
```

Comment:
When you use a `ref`, you don't add it as a dependency?

Reply:
I was wondering myself. I changed it to `useRef` because the lint rule complained about modifying `label` in the hook, recommending to use `useRef` instead.

https://github.com/facebook/react/issues/14387 seems to have an answer, especially https://github.com/facebook/react/issues/14387#issuecomment-455415008 and https://github.com/facebook/react/issues/14387#issuecomment-493677168

Reply:
Ah, looking at the code in more detail, is just a local variable in the function. The first reference (the early abort) should just be `label` whereas the others should be to a local `let newLabel = label` that might change during the function. And then still add it to the dependencies.

Reply:
Makes sense 👍